ui/desktop/src/components/SpellCheckContextMenu.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect, useRef } from 'react';
+
+interface SpellCheckContextMenuProps {
+  isOpen: boolean;
+  position: { x: number; y: number };
+  suggestions: string[];
+  misspelledWord: string;
+  onSuggestionSelect: (suggestion: string) => void;
+  onAddToDictionary: () => void;
+  onIgnore: () => void;
+  onClose: () => void;
+}
+
+export const SpellCheckContextMenu: React.FC<SpellCheckContextMenuProps> = ({
```

Comment:
We get this for free from the system spellchecker. Shouldn't in theory need an app level one.